tests/test_qp.py

```diff
@@ -22,12 +22,11 @@ class QPTest(PyverbsAPITestCase):
     Test various functionalities of the QP class.
     """
 
-    def test_create_qp_no_attr_connected(self):
+    def test_create_rc_qp_no_attr(self):
         """
-        Test QP creation via ibv_create_qp without a QPAttr object proivded.
-        Checked QP types are RC and UC.
+        Test RC QP creation via ibv_create_qp without a QPAttr object proivded.
         """
-        for ctx, attr, attr_ex in self.devices:
```

Comment:
consider using one "config" (one device, instead of iterating on all devices. splitting the tests and running on all devices will increase the run time).

Reply:
This is already handled by the PyverbsAPITestCase() class under the setUp() function.

Reply:
Nope. In case the user did not pass --dev, PyverbsAPITestCase() will add all devices (from ibv_get_device_list) to self.devices.
Which will cause each one of these tests to be run over each device/port.
On the other hand, these tests are basic, which should not take long (after I checked, all of them finish in up to a second even with your split).

Reply:
Isn't this the whole idea from all the tests that inherits from PyverbsAPITestCase(), I mean to go over all the available devices and run tests unless the user pass --dev.

Do you suggest a new design?

Reply:
My concern was that the tests start to take much longer because of multi-device runs.
BTW all traffic tests (which inherits from RDMATestCase) randomly choose one "config".

But I'll take this back, you can keep it as is for now. Initially, I thought that test_qp takes longer than the actual results.